cli/module_generate/scripts/generate_stubs.go

```diff
@@ -118,28 +117,42 @@ func setGoModuleTemplate(clientCode string, module modulegen.ModuleInputs) (*mod
 	return &goTmplInputs, nil
 }
 
-// formatType outputs typeExpr as readable string.
-func formatType(typeExpr ast.Expr) string {
+// formatType formats typeExpr as readable string with correct attribution if applicable.
+func formatType(typeExpr ast.Expr, resourceSubtype string) string {
 	var buf bytes.Buffer
 	err := printer.Fprint(&buf, token.NewFileSet(), typeExpr)
 	if err != nil {
 		return fmt.Sprintf("Error formatting type: %v", err)
 	}
-	return buf.String()
-}
-
-func handleMapType(str, resourceSubtype string) string {
-	endStr := strings.Index(str, "]")
-	keyType := strings.TrimSpace(str[4:endStr])
-	valueType := strings.TrimSpace(str[endStr+1:])
-	if unicode.IsUpper(rune(keyType[0])) {
-		keyType = fmt.Sprintf("%s.%s", resourceSubtype, keyType)
+	typeString := buf.String()
+
+	prefixes := []string{"*", "[]*", "[]", "chan "}
+	// checkUpper attributes to <resourceSubtype> if type is capitalized after prefix.
```

Comment:
(minor) I think there's a word missing between "checkUpper" and "attributes"? Either that or i'm just having trouble parsing the grammar here.